### PR TITLE
PLAT-882: Fix cross browser issues

### PR DESCRIPTION
**Pull request by f-jauregui**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-882)

#### Description
All detected cross browser and OS component issues in storybook were solved in this little PR.
This also includes a change in the zen-checkbox box element that might solve [this issue](https://reciprocitylabs.atlassian.net/browse/BAN-97), by unifying all browsers checkbox with an SVG and added the **indeterminate** variant.

#### ChangeLOG

- [ ] I have added all notable deployment/development environment (onprem!) changes to [CHANGELOG.md](https://github.com/reciprocity/zen-ui/blob/main/README.md)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

File: src/components/zen-button/zen-button.scss [hunk 1]
```diff
@@ -4,6 +4,7 @@
 :host {
   display: flex;
   width: fit-content;
+  width: -moz-fit-content;
```
Comment (mplesec) [inline src/components/zen-button/zen-button.scss:7]: Might make sense to just add https://github.com/browserslist/browserslist 😅 Fewer of these one off fixes

File: src/components/zen-checkbox/zen-checkbox.scss [hunk 2]
```diff
@@ -25,16 +26,58 @@
...
+  }
+
+  &:checked {
+    background-color: $color-blue-700;
+    border: 2px solid $color-blue-700;
+    &:disabled {
+      background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='12' height='9' viewBox='0 0 12 9'%3E%3Cpath d='M10.2188 0.445312L4.05469 6.63281L1.78125 4.35938C1.73438 4.29688 1.67188 4.26562 1.59375 4.26562C1.51562 4.26562 1.44531 4.29688 1.38281 4.35938L0.726562 5.01562C0.664062 5.07812 0.632812 5.14844 0.632812 5.22656C0.632812 5.30469 0.664062 5.36719 0.726562 5.41406L3.84375 8.55469C3.90625 8.60156 3.97656 8.625 4.05469 8.625C4.13281 8.625 4.19531 8.60156 4.24219 8.55469L11.2734 1.52344C11.3359 1.46094 11.3672 1.39062 11.3672 1.3125C11.3672 1.23438 11.3359 1.17188 11.2734 1.125L10.6172 0.445312C10.5547 0.398438 10.4844 0.375 10.4062 0.375C10.3281 0.375 10.2656 0.398438 10.2188 0.445312Z' fill='lightgrey'/%3E%3C/svg%3E");
```
Comment (mplesec) [inline src/components/zen-checkbox/zen-checkbox.scss:58]: Hm, why these svgs?

**Review by mplesec — APPROVED** [review]
Good work 👍

File: src/components/zen-checkbox/zen-checkbox.scss (hunk 2 above)
Comment (f-jauregui) [reply src/components/zen-checkbox/zen-checkbox.scss:58]: Depending the browser and also the OS these checks "icons" and boxes are different. In this way we create an homogeneous design across all browsers and OS.
SVG is the lightest way to do this. Bootstrap 5 for example is taking this approach: https://getbootstrap.com/docs/5.0/forms/checks-radios/

File: src/components/zen-tooltip/zen-tooltip.scss [hunk 3]
```diff
@@ -4,7 +4,6 @@ $arrow_size: 0.3rem;
 
 :host {
   position: absolute;
-  max-width: 11rem;
```
Comment (chavlji) [inline src/components/zen-tooltip/zen-tooltip.scss:7]: Why should this be removed? Long tooltips should be broken into multiple lines. This looks strange:
Screenshot 2021-01-27 at 15 38 18

File: src/components/zen-checkbox/zen-checkbox.scss [hunk 4]
```diff
@@ -25,16 +25,44 @@
...
+  background-repeat: no-repeat;
+  background-position: center;
+  background-size: contain;
+  transition: background-color $transition-element-bump, background-position $transition-element-bump,
+    border-color $transition-element-bump, box-shadow $transition-element-bump;
+  border-radius: 2px;
+  cursor: pointer;
+
+  &:checked[type='checkbox'] {
+    background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='12' height='9' viewBox='0 0 12 9' fill='none'%3E%3Cpath d='M10.2188 0.445312L4.05469 6.63281L1.78125 4.35938C1.73438 4.29688 1.67188 4.26562 1.59375 4.26562C1.51562 4.26562 1.44531 4.29688 1.38281 4.35938L0.726562 5.01562C0.664062 5.07812 0.632812 5.14844 0.632812 5.22656C0.632812 5.30469 0.664062 5.36719 0.726562 5.41406L3.84375 8.55469C3.90625 8.60156 3.97656 8.625 4.05469 8.625C4.13281 8.625 4.19531 8.60156 4.24219 8.55469L11.2734 1.52344C11.3359 1.46094 11.3672 1.39062 11.3672 1.3125C11.3672 1.23438 11.3359 1.17188 11.2734 1.125L10.6172 0.445312C10.5547 0.398438 10.4844 0.375 10.4062 0.375C10.3281 0.375 10.2656 0.398438 10.2188 0.445312Z' fill='white'/%3E%3C/svg%3E");
```
Comment (chavlji) [inline src/components/zen-checkbox/zen-checkbox.scss:45]: Could we use standard fa icons for this purpose? Just for the sake of standardization.
We've already used fa svgs on steps component, notifications,...
You can add a new task for this that will be done when we have the icon task ready.

**Review by chavlji — CHANGES_REQUESTED** [review]
Great job! Please just restore the tooltip's max-width.

**Review by chavlji — APPROVED** [review]

**Pull request merged**